objcgen/NameGenerator.cs

```diff
@@ -146,6 +149,32 @@ public static string GetMonoName (Type t)
 			}
 		}
 
+		public static string GetArrayTypeName (Type t)
+		{
+			switch (Type.GetTypeCode (t)) {
+			case TypeCode.Boolean:
+			case TypeCode.Char:
+			case TypeCode.Double:
+			case TypeCode.Single:
+			case TypeCode.SByte:
+			case TypeCode.Int16:
+			case TypeCode.Int32:
+			case TypeCode.Int64:
+			case TypeCode.UInt16:
+			case TypeCode.UInt32:
+			case TypeCode.UInt64:
+				return "NSArray <NSNumber *>";
+			case TypeCode.Byte:
+				return "NSData";
+			case TypeCode.String:
+				return "NSArray <NSString *>";
+			case TypeCode.Object:
+				return $"NSArray<{GetObjCName (t)} *>";
+			default:
+				throw new NotImplementedException ($"Converting type {t.Name} to a native type name");
```

Comment:
A `99` is something that should not happen (or be reached) and for which we need a test case.

A `NotImplementedException` message is for incomplete code, the exception message should be enough (without a test case).